s3-encryption/client.md

```diff
@@ -0,0 +1,142 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Client
+
+## Version
+
+0.1.0
+
+### Changelog
+
+- 0.1.0
+  - Initial
+
+## Definitions
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC2119](https://tools.ietf.org/html/rfc2119).
+
+### Key Material
+
+An AES or RSA key, or a valid AWS KMS key identifier.
+
+### Client Initialization
+
+Client initialization refers to the process by which a new S3EC instance is created.
+Depending on the language, this could be a builder (using the Builder pattern) or a constructor.
```

Comment:
perhaps remove? afaik builder design pattern is primarily used for OOP - so maybe this is more of an implementation detail? 
```suggestion
```

Reply:
sure, I tend to prefer being thorough and including conditional guidance, but you're right that this is an implementation detail. (Builder pattern may well go out of style eventually in OOP languages)